tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op.cc

```diff
@@ -677,7 +685,8 @@ void TRTEngineOp::ComputeAsync(OpKernelContext* ctx,
       return;
     } else if (cache_res->profiles_.GetNumProfiles() == 0) {
       // Create profiles out of collected shapes during profile generation.
-      cache_res->profiles_.InitProfiles();
+      cache_res->profiles_.AddShape(input_concrete_shapes, true);
```

Comment:
Shall we at least skip AddShape if HasShape(cache_res->profiles_)? Otherwise, we always add such an additional shape during inference, which is not what we want. If we will make this change, do you need to adjust some of the tests?

Reply:
The second arg (skip_if_has_shapes=true) of AddShape does that. Let me know if you prefer to handle this check differently.

I have adjusted trt_engine_op_test.cc, see around lines 250 and 280.

Reply:
I see. It is probably more readable to add a condition here. Instead of modifying AddShape, you may need to export HasShape from the profile class.

Reply:
I have changed it according to your suggestion.